### PR TITLE
BFB-212

### PR DESCRIPTION
**Pull request by wbeltran777**
**JIRA Ticket**
https://perigongames.atlassian.net/browse/BFB-212

**Description**
Added simple fmod mixer for controlling SFX and Music

### CODE REVIEW

**Comment by unoctium1** [comment]
@wbeltran777 Is the Music VCA still in use?

File: Assets/_BForBoss/_UserInterface/Scripts/VolumeSliderBehaviour.cs [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+        private void HandleOnSliderValueChanged(float value)
+        {
+            CustomInputField.text = value.ToString("F0");
+            SetVolume(SliderValue);
+        }
+
+        private void HandleOnInputFieldEnded(string value)
+        {
+            if (String.IsNullOrWhiteSpace(value))
```
Comment (Gursaanj) [inline Assets/_BForBoss/_UserInterface/Scripts/VolumeSliderBehaviour.cs:122]: should this be `string.IsNullOrEmpty(volume)` ?

File: Assets/_BForBoss/_UserInterface/Scripts/VolumeSliderBehaviour.cs [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+                CustomInputField.text = CustomSlider.value.ToString("F0");
+            }
+            else
+            {
+                CustomSlider.value = float.Parse(value);
+                SetVolume(SliderValue);
+            }
+        }
+        
+        private enum VolumeType
```
Comment (Gursaanj) [inline Assets/_BForBoss/_UserInterface/Scripts/VolumeSliderBehaviour.cs:133]: Not sure if this is still our coding convention but I think we place contained classes and enums above everything else (i.e. fields and stuff)

File: Assets/_BForBoss/_UserInterface/Scripts/VolumeSliderBehaviour.cs [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+        {
+            VCA, Bus, MasterBus
+        }
+    }
+
+    public interface VolumeWrapper
```
Comment (Gursaanj) [inline Assets/_BForBoss/_UserInterface/Scripts/VolumeSliderBehaviour.cs:139]: should be renamed `IVolumeWrapper`. Would we want to put this in its own separate script?

File: Assets/_BForBoss/_UserInterface/Scripts/VolumeSliderBehaviour.cs [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+        
+        public BusVolume(string name)
+        {
+            _bus = RuntimeManager.GetBus(BUS_PREFIX + name);
+        }
+
+        public void SetVolume(float volume)
+        {
+            if (!_bus.isValid())
+                return;
```
Comment (Gursaanj) [inline Assets/_BForBoss/_UserInterface/Scripts/VolumeSliderBehaviour.cs:187]: missing Curly Braces

File: Assets/_BForBoss/_Core/Scripts/UserInterface/AudioSettingsViewBehaviour.cs [hunk 5]
```diff
@@ -0,0 +1,13 @@
+using FMODUnity;
```
Comment (Gursaanj) [inline Assets/_BForBoss/_Core/Scripts/UserInterface/AudioSettingsViewBehaviour.cs:1]: Is this class necessary at the moment?

File: Assets/_BForBoss/_Core/Scripts/UserInterface/AudioSettingsViewBehaviour.cs [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+using UnityEngine;
+
+namespace BForBoss
```
Comment (Gursaanj) [inline Assets/_BForBoss/_Core/Scripts/UserInterface/AudioSettingsViewBehaviour.cs:4]: Maybe we should have a `BForBoss.Audio` Namespace? What do you think @unoctium1

File: Assets/_BForBoss/_UserInterface/Scripts/VolumeSliderBehaviour.cs (hunk 1 above)
Comment (unoctium1) [reply Assets/_BForBoss/_UserInterface/Scripts/VolumeSliderBehaviour.cs]: This included a bunch of WIP code from my changes, separate to what Will was working on. Just reverted my commit to clean up the PR

**Comment by unoctium1** [comment]
@wbeltran777 for the conflicts, I merged in the temp version of the mixers just so that we could get a working and testable version of the audio settings. You should be good to overwrite all the files though when you merge

**Review by unoctium1 — APPROVED** [review]
lgtm, sorry for all the hassle!

**Pull request merged**